kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/rbac/extractor/OauthAuthorityExtractor.java

```diff
@@ -25,7 +49,71 @@ public Mono<Set<String>> extract(AccessControlService acs, Object value, Map<Str
       throw new RuntimeException();
     }
 
-    return Mono.just(Set.of(principal.getName())); // TODO #2844
+    var provider = (OAuthProperties.OAuth2Provider) additionalParams.get("provider");
+    var rolesFieldName = provider.getCustomParams().get(ROLES_FIELD_PARAM_NAME);
```

Comment:
provider & rolesFieldName can be null?

Reply:
`provider` is always filled in `OAuthSecurityConfig`. Added an assert to ensure that.
`rolesFieldName` is also present, checked within `isApplicable` method, otherwise `extract` wouldn't be called.